### PR TITLE
Native OVS feature page add limitations

### PR DESCRIPTION
**Pull request by almusil**
@dominikholler @dankenigsberg @EdDev

### CODE REVIEW

**Comment by dankenigsberg** [comment]
Please put limitations right under Overview. Limitations are more important than implementation details.
Also, "upgrade" is not supported, and so are connection of a VM directly to OvS bridge. We support only OVN for that. Please update the whole feature page to match current reality.

**Comment by almusil** [comment]
I have updated the paragraph about upgrade from native to ovs as it is not currently supported. I did not find anything else that is in different state now.

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 1]
```diff
@@ -46,6 +46,20 @@ In the following table, networking entities are listed with their implementation
 ### OVS networking model 
 ![OVS networking model](/feature/openvswitch/ovs_networking_model.svg)
 
+## Limitations
+
+Currently there are some limitations compared to Linux bridge [RFE OvS support](https://bugzilla.redhat.com/show_bug.cgi?id=1195208):
+
+- [Multiple gateways](https://bugzilla.redhat.com/show_bug.cgi?id=1383035) - Source routing is not implemented for the OVS switch type. This means that only management network can have gateway.
+- DHCPv6 - Currently not supported
+- [Port mirroring](https://bugzilla.redhat.com/show_bug.cgi?id=1362492)
```
Comment (dankenigsberg) [inline source/feature/openvswitch/native-openvswitch.html.md]: please state that this should be solved in OVN

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 2]
```diff
@@ -46,6 +46,20 @@ In the following table, networking entities are listed with their implementation
...
+- [Host QoS](https://bugzilla.redhat.com/show_bug.cgi?id=1380271)
+- [Setting bridge options](https://bugzilla.redhat.com/show_bug.cgi?id=1380273)
+- [iSCSI Bond](https://bugzilla.redhat.com/show_bug.cgi?id=1441245)
+- LLDP support
+- Setting DNS
+- Upgrade from Linux bridge to OvS and vice versa
+- Direct connection of VM to the network - Only supported with OVN
```
Comment (dankenigsberg) [inline source/feature/openvswitch/native-openvswitch.html.md]: this is not a short-term limitation, it is an issue that would not change and deserves being mentioned in the Overview.

File: source/feature/openvswitch/native-openvswitch.html.md (hunk 1 above)
Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done

File: source/feature/openvswitch/native-openvswitch.html.md (hunk 2 above)
Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 3]
```diff
@@ -25,6 +25,7 @@ OVS is to be fully integrated (natively) into oVirt through VDSM, becoming the d
 ## Overview
 The introduction of OVS as an additional network switching solution raises the need to generalize and abstract the requirements from the system networking services and define a spec which considers both options (OVS & native Linux networking).
 The OVS model of arranging networking entities is different from the native Linux networking, but is closer to industry networking modeling.
+Direct connection of VM to the network cannot be done by OvS but it is supported only with OVN.
```
Comment (dankenigsberg) [inline source/feature/openvswitch/native-openvswitch.html.md]: This is too succinct. Please try something along the following lines - after all, autodefine is *your* feature!

Direct connection of VM to the network cannot be done by OvS. In order to connect a VM to an OvS bridge, a user must first define an OvS VM network. This automatically defines an OVN network that. The VM can be attached to the OVN network which is based on the OvS one.

Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done. I have hopefully described some of the missing features in more user friendly way.

**Comment by dankenigsberg** [comment]
@mmirecki I think you can take this in.

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 4]
```diff
@@ -25,6 +25,8 @@ OVS is to be fully integrated (natively) into oVirt through VDSM, becoming the d
 ## Overview
 The introduction of OVS as an additional network switching solution raises the need to generalize and abstract the requirements from the system networking services and define a spec which considers both options (OVS & native Linux networking).
 The OVS model of arranging networking entities is different from the native Linux networking, but is closer to industry networking modeling.
+Direct connection of VM to the network cannot be done by OvS. In order to connect a VM to an OvS bridge, a user must first define an OvS VM network.
```
Comment (mmirecki) [inline source/feature/openvswitch/native-openvswitch.html.md]: "Direct connection of VM to the network cannot be done by OvS"
This is not quite clear.
We can connect a VM to a network (ovirt VM network, represented by a ovs bridge).

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 5]
```diff
@@ -46,6 +48,19 @@ In the following table, networking entities are listed with their implementation
 ### OVS networking model 
 ![OVS networking model](/feature/openvswitch/ovs_networking_model.svg)
 
+## Limitations
+
+Currently there are some limitations compared to Linux bridge [RFE OvS support](https://bugzilla.redhat.com/show_bug.cgi?id=1195208):
+
+- [Multiple gateways](https://bugzilla.redhat.com/show_bug.cgi?id=1383035) - Source routing is not implemented for the OVS switch type. This means that only management network can have gateway.
+- DHCPv6 - Dynimac address configuration for IPv6 is currently not supported.
```
Comment (mmirecki) [inline source/feature/openvswitch/native-openvswitch.html.md]: Dynimac -> dynamic

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 6]
```diff
@@ -46,6 +48,19 @@ In the following table, networking entities are listed with their implementation
...
 ![OVS networking model](/feature/openvswitch/ovs_networking_model.svg)
 
+## Limitations
+
+Currently there are some limitations compared to Linux bridge [RFE OvS support](https://bugzilla.redhat.com/show_bug.cgi?id=1195208):
+
+- [Multiple gateways](https://bugzilla.redhat.com/show_bug.cgi?id=1383035) - Source routing is not implemented for the OVS switch type. This means that only management network can have gateway.
+- DHCPv6 - Dynimac address configuration for IPv6 is currently not supported. 
+- [Port mirroring](https://bugzilla.redhat.com/show_bug.cgi?id=1362492) -  Should be done via OVN.
+- [Host QoS](https://bugzilla.redhat.com/show_bug.cgi?id=1380271) - OvS supports different parameters to configure the limitations. Which is currently incompatible with solution used for Linux bridge.
```
Comment (mmirecki) [inline source/feature/openvswitch/native-openvswitch.html.md]: OvS supports different parameters to configure the limitations. Which is currently incompatible with solution used for Linux bridge.
->
 OvS supports different parameters for configuring QoS, Which are incompatible with the ones used for a Linux bridge.

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 7]
```diff
@@ -104,31 +119,34 @@ Minimal changes in current API verbs are required:
...
-- Receive a setupNetworks call.
-- Detect that a network has been marked with a type that is different from the current.
-- Validate that all existing networks exist in the command and that all are marked with the same type.
-  - We would like to perform an atomic upgrade, and not to worry about codependent networks with differing switch types. 
-- Tear down the pre-existing networks and bondings..
-- Create networks.
-
-Note: Engine user may decide to take the reverse transition, from OVS to legacy, as well.
+
+Upgrade is currently not supported. Causing networks to go out-of-sync due to differences between native and OvS see [Limitations](#limitations). Only newly added hosts can be configured with OvS networking. Mixing native and OvS in cluster is not supported. Plase note that the change is blocked only in UI but is still possible via REST api.
```
Comment (mmirecki) [inline source/feature/openvswitch/native-openvswitch.html.md]: Upgrade is currently not supported. Causing networks to go out-of-sync due to differences between native and OvS see [Limitations](#limitations).
->
Upgrade from native to OVS networking is currently not supported. An upgrade will cause networks to be out-of-sync due to differences between native and OvS (see Limitations).

Comment (mmirecki) [inline source/feature/openvswitch/native-openvswitch.html.md]: Mixing native and OvS in cluster is not supported.
->
Mixing native and OvS networks in a cluster is not supported.

Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done

Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done

File: source/feature/openvswitch/native-openvswitch.html.md (hunk 6 above)
Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done

File: source/feature/openvswitch/native-openvswitch.html.md (hunk 5 above)
Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done

File: source/feature/openvswitch/native-openvswitch.html.md (hunk 4 above)
Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: I have added autodefine feature page to make it more clear hopefully

Comment (mmirecki) [inline source/feature/openvswitch/native-openvswitch.html.md]: Maybe reword:
Direct connection of VM to the network cannot be done by OvS. -> A VM can not be simply connected to an OvS bridge.

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 8]
```diff
@@ -25,6 +25,8 @@ OVS is to be fully integrated (natively) into oVirt through VDSM, becoming the d
 ## Overview
 The introduction of OVS as an additional network switching solution raises the need to generalize and abstract the requirements from the system networking services and define a spec which considers both options (OVS & native Linux networking).
 The OVS model of arranging networking entities is different from the native Linux networking, but is closer to industry networking modeling.
+Direct connection of VM to the network cannot be done by OvS. In order to connect a VM to an OvS bridge, a user must first define an OvS VM network. 
+This automatically creates an OVN network, which is connected to OvS one. The OVN network can then be attached to a VM (see [Autodefine External Network](https://ovirt.org/develop/release-management/features/network/autodefine-external-network/)).
```
Comment (mmirecki) [inline source/feature/openvswitch/native-openvswitch.html.md]: "which is connected to OvS one"
The OVN network is not connected, but rather built on top of OvS.

"The OVN network can then be attached to a VM"
The VM is still connected to the intergration OvS bridge, with an OVN overlay.

Maybe something like (please feel free to improve further):
This automatically creates an OVN network, which is connected to OvS one. The OVN network can then be attached to a VM (see [Autodefine External Network](https://ovirt.org/develop/release-management/features/network/autodefine-external-network/)).
==>
This automatically creates an OVN network on top of the OvS integration bridge. Te VM is is then attached to the OVN network via the OvS integration bridge (see [Autodefine External Network](https://ovirt.org/develop/release-management/features/network/autodefine-external-network/)).

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 9]
```diff
@@ -46,6 +48,18 @@ In the following table, networking entities are listed with their implementation
 ### OVS networking model 
 ![OVS networking model](/feature/openvswitch/ovs_networking_model.svg)
 
+## Limitations
+
+Currently there are some limitations compared to Linux bridge [RFE OvS support](https://bugzilla.redhat.com/show_bug.cgi?id=1195208):
+
+- [Multiple gateways](https://bugzilla.redhat.com/show_bug.cgi?id=1383035) - Source routing is not implemented for the OVS switch type. This means that only management network can have gateway.
```
Comment (mmirecki) [inline source/feature/openvswitch/native-openvswitch.html.md]: gateway -> a gateway

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 10]
```diff
@@ -46,6 +48,18 @@ In the following table, networking entities are listed with their implementation
...
+- [Port mirroring](https://bugzilla.redhat.com/show_bug.cgi?id=1362492) -  Should be done via OVN.
+- [Host QoS](https://bugzilla.redhat.com/show_bug.cgi?id=1380271) - OvS supports different parameters for configuring QoS. Whicharey incompatible with the ones used for a Linux bridge.
```
Comment (mmirecki) [inline source/feature/openvswitch/native-openvswitch.html.md]: QoS. Whicharey
->
QoS, which are

File: source/feature/openvswitch/native-openvswitch.html.md [hunk 11]
```diff
@@ -104,31 +118,34 @@ Minimal changes in current API verbs are required:
...
-- Validate that all existing networks exist in the command and that all are marked with the same type.
-  - We would like to perform an atomic upgrade, and not to worry about codependent networks with differing switch types. 
-- Tear down the pre-existing networks and bondings..
-- Create networks.
-
-Note: Engine user may decide to take the reverse transition, from OVS to legacy, as well.
+
+Upgrade from native to OvS is currently not supported and vice versa. An upgrade will cause networks to be out-of-sync due to differences between native and OvS (see [Limitations](#limitations)). Only newly added hosts can be configured with OvS networking. Mixing native and OvS networks in a cluster is not supported. Plase note that the change is blocked only in UI but is still possible via REST api. 
+
+~~Transition between switch types must be supported for the host as a whole. Nevertheless, if the user has defined his own networking configuration which is not controlled by VDSM, it should coexist with VDSM controlled networks.~~
```
Comment (mmirecki) [inline source/feature/openvswitch/native-openvswitch.html.md]: "Transition between switch types must be supported for the host as a whole."
Does this mean we can only have one switch type on a host?
Can you please state this more cleary?

Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: This is the old statement that is no longer valid

File: source/feature/openvswitch/native-openvswitch.html.md (hunk 10 above)
Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done

File: source/feature/openvswitch/native-openvswitch.html.md (hunk 9 above)
Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done

File: source/feature/openvswitch/native-openvswitch.html.md (hunk 8 above)
Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done

File: source/feature/openvswitch/native-openvswitch.html.md (hunk 4 above)
Comment (almusil) [reply source/feature/openvswitch/native-openvswitch.html.md]: Done
